source/Plugins.h

```diff
@@ -33,18 +34,39 @@ struct Plugin {
 	std::string path;
 	// The about text, if any, of this plugin.
 	std::string aboutText;
+	// The version of this plugin, importanted if it has been installed over ES.
+	std::string version = "???";
 
 	// Whether this plugin was enabled, i.e. if it was loaded by the game.
 	bool enabled = true;
 	// The current state of the plugin.
 	bool currentState = true;
+	// If this plugin has been deleted.
+	bool removed = false;
 };
 
 
 
 // Tracks enabled and disabled plugins for loading plugin data or skipping it.
 // This object is updated by toggling plugins in the Preferences UI.
 class Plugins {
+public:
+
+	struct InstallData
+	{
+		std::string name;
+		std::string url;
+		std::string version;
+		std::string aboutText;
+		bool installed = false;
+		bool outdated = false;
+		InstallData(std::string name = "", std::string url = "", std::string version = "",
```

Comment:
dont strings default to being empty without doing anything?

Reply:
but not when they are a function parameter?

Reply:
wait I just realised, we gonna install data with no name or url or version? lol

Reply:
no but sometimes you want to create an empty object to fill up or sometimes you want to create an object with all info already inside